docs/HowToGuides/GettingStarted.md

```diff
@@ -365,78 +365,65 @@ whenever the heading is modified.
-  viewed as a means of leveraging the navigation, editing and debugging features
-  of the IDE in relation to the source code they wrap.
-
-* Create an empty Xcode project in the workspace, using the
+This workflow enables you to edit, build, run, and debug in Xcode. The
+following steps assume that you have already [built the toolchain with Ninja](#the-actual-build).
+
+> **Note**  
+> A seamless LLDB debugging experience requires that your `build-script`
+  invocation for Ninja is tuned to produce build rules for the
+  [debug variant](#debugging-issues) of the component you intend to debug.
+
+* Generate the Xcode project with `utils/build-script --swift-darwin-supported-archs "$(uname -m)" --xcode --clean`.
+  This can take a few minutes due to metaprogrammed sources that depend on LLVM
+  tools that are built from source.
+* Create an empty Xcode workspace.
+* Add `build/Xcode-*/swift-macosx-*/Swift.xcodeproj` to the workspace. This
+  project includes the sources for almost everything in the repository,
+  including the compiler, standard library and runtime. If you intend to work on
+  a compiler subcomponent that is written in Swift and has a `Package.swift`
+  file (e.g. `lib/ASTGen`), add the package directory to the workspace by
+  choosing *File > Add Files to "\<workspace name\>"*. Xcode will automatically
+  generate schemes for the package manifest.
+* Create an Xcode project inside the workspace, using the
   _External Build System_ template.
-* Add a target to the empty project, using the _External Build System_ template,
-  and name it after the Ninja target that you want to build (e.g. `swift-frontend`
-  is the compiler).
+* Create a target in the new project, using the _External Build System_
+  template.
 * In the _Info_ pane of the target settings, set
   * _Build Tool_ to the absolute path of the `ninja` executable (the output of
     `which ninja` on the command line)
-  * _Arguments_ to the Ninja target (e.g. `bin/swift-frontend`)
-  * _Directory_ to the absolute path of the build directory where the Ninja
-    target lives. For Swift targets such as the compiler or standard library,
-    this is the `build/Ninja-*/swift-macosx-*` directory.
-* Add a scheme for the target. In the drop-down menu, be careful not to mistake
-  your target for a similar one that belongs to a generated Xcode project.
-* > **Note**  
-  > Ignore this step if the target associates to a non-executable Ninja target
-    like `swift-stdlib`.
-
-  Adjust the _Run_ action settings of the scheme:
-  * In the _Info_ pane, select the _Executable_ built by the Ninja target from
-    the appropriate `bin` directory (e.g. `build/Ninja-*/swift-macosx-*/bin/swift-frontend`).
-  * In the _Arguments_ pane, add the command line arguments that you want to
-    pass to the executable on launch (e.g. `path/to/file.swift -typecheck` for
+  * _Arguments_ to a Ninja target (e.g. `bin/swift-frontend`, the compiler)
```

Comment:
maybe some notes about each compiler so new developers can know how to which argument to pass to build pieces of their code?

Reply:
when making changes to any piece of code, do you have to rebuild the project everytime? What would you have to run in order to build the `lib/ASTGen` folder if you make changes to a `swift` file?

Reply:
> maybe some notes about each compiler so new developers can know how to which argument to pass to build pieces of their code?

You mean each target? That is definitely a future direction, yes!

Reply:
yes, my main question is what does the `swift-frontend` target even compile? Is that appropriate to rebuild pieces of code made in the `swiftAST`?

Reply:
> What would you have to run in order to build the `lib/ASTGen` folder if you make changes to a swift file?

`lib/ASTGen`, as anything else in the `lib` directory, is part of the compiler, so you should rebuild the compiler (the product) in order for your changes to come into force. If you just want to build ASTGen for diagnostics, you can use the scheme that Xcode generated for its package manifest (granted you added the package correctly).

> when making changes to any piece of code, do you have to rebuild the project everytime?

Certainly not. You could go as far as create targets and schemes for individual libraries, but most of the time you want to aim big and rebuild a product like the compiler or standard library, depending on what you’re working on. Builds are incremental, so rebuilding composite targets ought not take much longer than rebuilding the components unless e.g. you modify a heavily included C++ header.

Reply:
> granted you added the package correctly

I added the `ASTGen` folder that contains the `Package.swift` file, `Sources` folders, `LLVMJSON` folder, and `CMakeLists` file. It created a `package dependency` called `swift-syntax local` if that is what you are talking about. I do not see any schemes that were generated.

Reply:
Yes, correct. 

> I do not see any schemes that were generated.

I just tried creating an empty workspace and adding `lib/ASTGen` and this is what I see:

<img width="755" alt="Screen Shot 2023-09-17 at 00 52 45" src="https://github.com/apple/swift/assets/17835512/e40f77dc-ddd3-4e1b-83a4-033c8380e111">

Could it be hidden among all those schemes you autocreated?

Reply:
i had to autocreate the schemes and then it showed up

Reply:
> i had to autocreate the schemes and then it showed up

So you had *Autocreate schemes* unchecked in the scheme manager?

Reply:
so running this scheme is like "compiling" the files?

Reply:
> > i had to autocreate the schemes and then it showed up
> 
> So you had _Autocreate schemes_ unchecked in the scheme manager?

yes, I did

Reply:
> so running this scheme is like "compiling" the files?

Yes, it will build the corresponding target from the package manifest. FYI builds with this scheme do not contribute to the build artifacts in `swift-macosx-*/` because the scheme is generated by Xcode and the build artifacts will be written to the derived data. This target is merely a convenience that allows you to spot-build ASTGen in isolation from the rest of the compiler.

Reply:
> my main question is what does the `swift-frontend` target even compile

The compiler (includes ASTGen), as mentioned in the diff line this thread is attached to.

Reply:
Are any other executables important to know about when developing?

Reply:
> > So you had _Autocreate schemes_ unchecked in the scheme manager?
> 
> yes, I did

Thanks, will mention this caveat.